coral-schema/src/main/java/com/linkedin/coral/schema/avro/RelDataTypeToAvroType.java

```diff
@@ -52,7 +52,15 @@ static Schema relDataTypeToAvroTypeNonNullable(@Nonnull RelDataType relDataType,
     Preconditions.checkNotNull(relDataType);
 
     if (relDataType instanceof RelRecordType || relDataType instanceof DynamicRecordType) {
-      return relRecordTypeToAvroType(relDataType, null, recordName, "rel_avro", null);
+      String uniqueNamespace;
+      if (relDataType.getFieldList() == null || relDataType.getFieldCount() == 0) {
+        uniqueNamespace = "rel_avro";
+      } else {
+        uniqueNamespace =
+            "namespace_from_first_nested_field_" + relDataType.getFieldList().get(0).getName().toLowerCase();
```

Comment:
It looks like this naming pattern just adds the name of the first field in the struct as the suffix of the name space, so it seems cannot cover the corner case like 
```
struct f1 {
  struct f1 {
      struct f1
   }
}
```
all the fields are the first field in the struct in each level, so all the value of the name space still are the same like "namespace_from_first_nested_field_f1"
Please correct me if my understanding is wrong.

Reply:
+1 Not clear what the intention of the naming convention is.

Reply:
It doesnt cover the corner case that you mentioned @yiqiangin. However, its the minimum code changes needed to generate the schema successfully for the specified corner case.

Reply:
Can we make it more general? It is fine if it needs more code.